src/Microsoft.DocAsCode.Build.Engine/DocumentBuildParameters.cs

```diff
@@ -10,47 +10,47 @@ namespace Microsoft.DocAsCode.Build.Engine
 
     public sealed class DocumentBuildParameters : MarshalByRefObject
     {
+        [IncrementalIgnore]
         public FileCollection Files { get; set; }
 
+        [IncrementalIgnore]
         public string OutputBaseDir { get; set; }
 
-        [IncrementalCheck]
         public ImmutableArray<string> ExternalReferencePackages { get; set; } = ImmutableArray<string>.Empty;
 
-        [IncrementalCheck]
         public ImmutableArray<string> XRefMaps { get; set; } = ImmutableArray<string>.Empty;
 
-        [IncrementalCheck]
         public ImmutableDictionary<string, object> Metadata { get; set; } = ImmutableDictionary<string, object>.Empty;
 
-        [IncrementalCheck]
         public FileMetadata FileMetadata { get; set; }
 
-        [IncrementalCheck]
         public ImmutableArray<string> PostProcessors { get; set; } = ImmutableArray<string>.Empty;
 
+        [IncrementalIgnore]
         public TemplateManager TemplateManager { get; set; }
 
-        // todo : partial properties should check.
-        //[IncrementalCheck]
+        [IncrementalIgnore]
         public ApplyTemplateSettings ApplyTemplateSettings { get; set; }
 
+        [IncrementalIgnore]
         public int MaxParallelism { get; set; }
 
-        [IncrementalCheck]
         public string MarkdownEngineName { get; set; } = "dfm";
 
-        [IncrementalCheck]
         public ImmutableDictionary<string, object> MarkdownEngineParameters { get; set; } = ImmutableDictionary<string, object>.Empty;
 
+        [IncrementalIgnore]
         public string VersionName { get; set; }
 
         public string RootTocPath { get; set; }
 
+        [IncrementalIgnore]
         public string TemplateDir { get; set; }
 
+        [IncrementalIgnore]
         public ImmutableDictionary<string, ChangeKindWithDependency> Changes { get; set; }
 
+        [IncrementalIgnore]
```

Comment:
>IncrementalIgnore [](start = 9, length = 17)

when force rebuild, should we always full build? What happens when IncrementalIgnore this property?